### PR TITLE
docs: add  chatdataset

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog

- Add specific line by line info of high level changes in this PR.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: docs/guides/llm/chat-dataset.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+---
+
+## End-to-End Example: MASSIVE NLU
+
+The repository includes a complete worked example that fine-tunes Llama-3.2-3B-Instruct on the [AmazonScience/MASSIVE](https://huggingface.co/datasets/AmazonScience/massive) NLU dataset for intent detection and entity extraction.
+
+**Step 1** — Prepare the data (converts MASSIVE to OpenAI messages JSONL):
+
+```bash
+python examples/llm_finetune/prepare_massive_nlu.py --output_dir data/massive_nlu
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md:187]: **Critical**: `examples/llm_finetune/prepare_massive_nlu.py` does not exist in this branch. This step will fail for any user following the guide.

Is this script being added in a follow-up commit, or should the section reference an existing example instead?

File: docs/guides/llm/chat-dataset.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+**Step 2** — Launch training:
+
+```bash
+automodel finetune llm -c examples/llm_finetune/llama3_2/llama_3_2_3b_instruct_massive_nlu_peft.yaml
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md]: **Critical — two issues on this line:**

1. `examples/llm_finetune/llama3_2/llama_3_2_3b_instruct_massive_nlu_peft.yaml` does not exist in this branch. The end-to-end example is broken.

2. The CLI format `automodel finetune llm -c ...` is being replaced by `automodel <config.yaml>` in PR #1406. If that PR merges first, this command will be stale on arrival.

```suggestion
automodel examples/llm_finetune/llama3_2/llama_3_2_3b_instruct_massive_nlu_peft.yaml
```

(The YAML config and prep script also need to be added to the repo for this section to work.)

File: docs/guides/llm/chat-dataset.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+See also the Qwen chat example config at `examples/llm_finetune/qwen/qwen2_5_7b_instruct_chat.yaml`.
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md]: The Qwen config exists but its internal comment block still references the old invocation pattern (`uv run recipes/llm_finetune/finetune.py --config ...`). Consider updating the config comment to match the current CLI usage, or noting the correct command here.

```suggestion
See also the Qwen chat example config at [`examples/llm_finetune/qwen/qwen2_5_7b_instruct_chat.yaml`](https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/llm_finetune/qwen/qwen2_5_7b_instruct_chat.yaml).
```

File: docs/guides/llm/chat-dataset.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  _target_: nemo_automodel.components.datasets.llm.ChatDataset
+  path_or_dataset_id: Salesforce/xlam-function-calling-60k
+  split: train
+  seq_length: 2048
+  start_of_turn_token: "<start_of_turn>"
+```
+
+### Answer-Only Loss
+
+To train the model only on assistant responses (masking user/system tokens in the loss), provide `start_of_turn_token` — the token that marks where the assistant response begins in the rendered chat template. This is model-specific:
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md:160]: **Critical — `start_of_turn_token` is dead code.**

`ChatDataset.__init__` accepts and stores `start_of_turn_token` (`chat_dataset.py:156`), but `__getitem__` (lines 167–189) **never passes it** to `format_chat_template()`. The `format_chat_template` function in `formatting_utils.py` has no `start_of_turn_token` parameter — answer-only loss masking actually works via the tokenizer's `{% generation %}` Jinja2 keyword or manual prompt/assistant splitting.

This entire section (the table of model-specific tokens and the YAML example below) documents a mechanism that has no effect. Users who follow this guidance will set the parameter and see no change in behavior.

Either:
- Wire `start_of_turn_token` through to `format_chat_template` in the code, **or**
- Rewrite this section to describe the actual `{% generation %}` mechanism

File: docs/guides/llm/chat-dataset.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+}
+```
+
+Store your samples as **one JSON object per line** in a `.jsonl` file.
+
+---
+
+## Quickstart
+
+The fastest way to try it out — point at a local JSONL file and print the first tokenized sample:
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md:101]: **High — missing prerequisites.** The Quickstart has no mention of:

1. Installing nemo-automodel and transformers (`pip install nemo-automodel`)
2. Hugging Face authentication — `meta-llama/Llama-3.2-3B-Instruct` is a **gated model** requiring `huggingface-cli login` or `HF_TOKEN`
3. How to create the sample data file

A user following this verbatim will hit errors at every step. Consider adding a brief prerequisites block before the code:

```suggestion
The fastest way to try it out — point at a local JSONL file and print the first tokenized sample:

> **Prerequisites**: `pip install nemo-automodel transformers` and, for gated models like Llama, run `huggingface-cli login` first. Save one or more of the JSON examples from the [Data Format](#data-format) section as a `.jsonl` file (one object per line).
```

File: docs/guides/llm/chat-dataset.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+ds = ChatDataset(
+    path_or_dataset_id="data/my_chat_data/train.jsonl",
+    tokenizer=tokenizer,
+    seq_length=2048,
+    padding="max_length",
+    truncation="longest_first",
+)
+
+sample = ds[0]
+print(sample.keys())  # dict_keys(['input_ids', 'labels', 'attention_mask'])
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md]: **Medium — output keys are inaccurate.** `format_chat_template()` in `formatting_utils.py:172-181` returns 4 keys, not 3. The actual output includes an internal `___PAD_TOKEN_IDS___` dict used by the collator.

```suggestion
print(sample.keys())  # dict_keys(['input_ids', 'labels', 'attention_mask', '___PAD_TOKEN_IDS___'])
```

File: docs/guides/llm/chat-dataset.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+The fastest way to try it out — point at a local JSONL file and print the first tokenized sample:
+
+```python
+from transformers import AutoTokenizer
+from nemo_automodel.components.datasets.llm import ChatDataset
+
+tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-3.2-3B-Instruct")
+
+ds = ChatDataset(
+    path_or_dataset_id="data/my_chat_data/train.jsonl",
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md]: **Medium — no instructions to create this file.** The Data Format section above shows JSON examples but never connects them to a usable `.jsonl` file. A copy-paste user will get `FileNotFoundError` here.

```suggestion
    path_or_dataset_id="data/my_chat_data/train.jsonl",  # save Data Format examples as one JSON object per line
```

File: docs/guides/llm/chat-dataset.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+  split: train
+  seq_length: 2048
+  start_of_turn_token: "<start_of_turn>"
+```
+
+### Answer-Only Loss
+
+To train the model only on assistant responses (masking user/system tokens in the loss), provide `start_of_turn_token` — the token that marks where the assistant response begins in the rendered chat template. This is model-specific:
+
+| Model family | `start_of_turn_token` |
```
Comment (chenopis) [inline docs/guides/llm/chat-dataset.md:162]: **Medium — table values cannot be verified.** Since `start_of_turn_token` is not wired into the loss masking path (see comment on line 158), these model-specific token values have no functional effect and cannot be validated through code. If the parameter is fixed in code, these values should be cross-checked against the actual tokenizer chat templates for each model family.
